modin/backends/base/query_compiler.py

```diff
@@ -143,11 +150,15 @@ def binary_op(self, op, other, **kwargs):
         Returns:
             A new QueryCompiler object.
         """
-        raise NotImplementedError(NOT_IMPLEMENTED_MESSAGE)
+        # TODO This doesn't seem to be used, children classes seem
+        # to implement the binary methods directly
+        return getattr(self, op)(other, **kwargs)
```

Comment:
Instead of `binary_op` let's enumerate them in the base class as well. The list can be found here: https://github.com/modin-project/modin/blob/9e21b589d0baaee3de4f31fe2171f3dfaa9363a7/modin/backends/pandas/query_compiler.py#L194-L220